### PR TITLE
Run tests with requested locale.

### PR DESCRIPTION
**Pull request by ilonatommy**
For https://github.com/dotnet/runtime/pull/80421/ we need tests to run with specific locale set, otherwise we won't be able to test the mechanism of loading different internationalization data based on the client's locale.
I added `-locale=xx-YY` flag that will set environment variable before the test on Browser / Node is launched. Default is "en-US".

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.CLI/CommandArguments/WASM/Arguments/LocaleArgument.cs
Comment (premun) [inline src/Microsoft.DotNet.XHarness.CLI/CommandArguments/WASM/Arguments/LocaleArgument.cs:10]: I think you're not using the `defaultValue` in any way? Have a look at `RequiredStringArgument` and pass it to its ctor.

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/JS/WasmTestCommand.cs
Comment (premun) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/JS/WasmTestCommand.cs:117]: It's also possible to set the env just for the process you're launching with `processManager` below. Maybe that's better? Or do you need this for other processes you don't control?

Comment (ilonatommy) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/JS/WasmTestCommand.cs:117]: Only for the process below, should get changed, thanks

File: src/Microsoft.DotNet.XHarness.CLI/CommandArguments/WASM/Arguments/LocaleArgument.cs
Comment (premun) [inline src/Microsoft.DotNet.XHarness.CLI/CommandArguments/WASM/Arguments/LocaleArgument.cs:11]: nit: I guess this text doesn't apply for node so we could make it more generic and not spell out the concrete variable?

E.g. 
```suggestion
        : base("locale=", $"Sets the browser's/node's language through an environment variable, default value {defaultValue}", defaultValue)
```

Comment (premun) [reply src/Microsoft.DotNet.XHarness.CLI/CommandArguments/WASM/Arguments/LocaleArgument.cs:11]: Also it's clear it's not setting XHarness' output language or similar

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs
Comment (premun) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:75]: This one stays?

Comment (ilonatommy) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:75]: For this one I haven't figured out the way yet, reading into it

**Review by premun — APPROVED** [review]
LGTM! But I'd suggest also getting a review from someone on the WASM team.

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs
Comment (pavelsavara) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:75]: There is `--lang` for chrome. Is that better ?

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs
Comment (pavelsavara) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs:71]: Why we are doing it on server side?

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs
Comment (ilonatommy) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:75]: It does not work on Linux, https://developer.chrome.com/docs/extensions/reference/i18n/#linux

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs
Comment (ilonatommy) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs:71]: That was the only idea I came up with to pack setting the LANGUAGE var for the testing process only. I don't see any way of setting it to the BrowserDriver because adding "--lang" argument on Linux it does not have any effect. In my understanding it's the equivalent of setting the env variable for the test process, isn't it?

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs
Comment (radical) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:75]: We have been using `--lang` with debugger tests - https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/ChromeProvider.cs#L166-L167 . Could you also confirm that it doesn't actually work there, and update if needed?

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/JS/WasmTestCommand.cs
Comment (radical) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/JS/WasmTestCommand.cs:143]: `wasm test` runs for v8 also, and will be running for other js engines too.

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs
Comment (ilonatommy) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:75]: So far it works because we are loading full ICU in debugger tests. As soon as we merge ICU sharding PR the line you quoted will be good for Windows but will fail with Linux. I already fixed it on that PR, it's in this commit: https://github.com/dotnet/runtime/pull/80421/commits/dde05ee00bf0aecacfbfe6e302f34f65e272e2d2.

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs
Comment (radical) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs:71]: If the browser process needs this environment variable then this needs to be set when launching chrome.

Comment (ilonatommy) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs:71]: Maybe I will paste my test screenshots here from Linux.
Not good:
image
Good:
image

Comment (radical) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs]: Set the environment variable on the process launch then.

**Review by pavelsavara — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs
Comment (ilonatommy) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs]: Credits @radical:
image

**Pull request merged**
